yaserde/tests/enum.rs

```diff
@@ -357,3 +357,43 @@ fn unnamed_enum() {
   serialize_and_validate!(model, content);
   deserialize_and_validate!(content, model, XmlStruct);
 }
+
+#[test]
+fn tagged_enum() {
+  #[derive(Debug, PartialEq, YaSerialize, YaDeserialize, Default)]
+  #[yaserde(tag = "type")]
+  enum FooBar {
+    #[default]
+    #[yaserde(rename = "foo")]
+    Foo,
+    #[yaserde(rename = "bar")]
+    Bar,
+  }
+
+  #[derive(Debug, PartialEq, YaSerialize, YaDeserialize, Default)]
+  #[yaserde(rename = "base")]
+  struct Base {
+    #[yaserde(rename = "foobar")]
+    foo_bar: FooBar,
+  }
+
+  let model = FooBar::Foo;
+  let content = "<base type=\"foo\" />";
+  deserialize_and_validate!(content, model, FooBar);
+
+  let model = FooBar::Bar;
+  let content = "<base type=\"bar\" />";
+  deserialize_and_validate!(content, model, FooBar);
+
+  let model = Base {
+    foo_bar: FooBar::Foo,
+  };
+  let content = "<base><foobar type=\"foo\" /></base>";
+  deserialize_and_validate!(content, model, Base);
+
+  let model = Base {
+    foo_bar: FooBar::Foo,
+  };
+  let content = "<base><foobar type=\"foo\" /></base>";
```

Comment:
The unit test complains at this point.

Serialization is not handled is your commits.
Can you complete it ?

Reply:
I have added basic serialization support. However, this made me realise that i am not sure what the behavior should be when combining tag / flatten.